### PR TITLE
[Merged by Bors] - feat(RingTheory/TwoSidedIdeal/Basic): two-sided-ideal as a bimodule

### PR DESCRIPTION
**Pull request by jjaassoonn**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/TwoSidedIdeal/Basic.lean [hunk 1]
```diff
@@ -169,6 +170,58 @@ instance addCommGroup : AddCommGroup I :=
...
+
+instance : SMul R I where smul r x := ⟨r • x.1, I.mul_mem_left _ _ x.2⟩
+
+instance : SMul Rᵐᵒᵖ I where smul r x := ⟨r • x.1, I.mul_mem_right _ _ x.2⟩
+
+instance leftModule : Module R I :=
+  Function.Injective.module _ (coeAddMonoidHom I) Subtype.coe_injective fun _ _ => rfl
+
+@[simp]
+lemma coe_smul {r : R} {x : I} : ↑(r • x : R) = r * (x : R) := rfl
```
Comment (kbuzzard) [inline Mathlib/RingTheory/TwoSidedIdeal/Basic.lean:194]:
```suggestion
lemma coe_smul {r : R} {x : I} : (r • x : R) = r * (x : R) := rfl
```
That coercion is from R to R. Similarly line 200.

File: Mathlib/RingTheory/TwoSidedIdeal/Basic.lean [hunk 2]
```diff
@@ -169,6 +170,58 @@ instance addCommGroup : AddCommGroup I :=
...
+section Ring
+
+variable {R : Type*} [Ring R] (I : TwoSidedIdeal R)
+
+instance : SMul R I where smul r x := ⟨r • x.1, I.mul_mem_left _ _ x.2⟩
+
+instance : SMul Rᵐᵒᵖ I where smul r x := ⟨r • x.1, I.mul_mem_right _ _ x.2⟩
+
+instance leftModule : Module R I :=
+  Function.Injective.module _ (coeAddMonoidHom I) Subtype.coe_injective fun _ _ => rfl
```
Comment (kbuzzard) [inline Mathlib/RingTheory/TwoSidedIdeal/Basic.lean:191]:
```suggestion
  Function.Injective.module _ (coeAddMonoidHom I) Subtype.coe_injective fun _ _ ↦ rfl
```
I think that's the mathlib style? Similarly line 197

**Review by kbuzzard — APPROVED** [review]
This looks fine to me. Thanks! I left some minor comments.

File: Mathlib/RingTheory/TwoSidedIdeal/Basic.lean (hunk 1 above)
Comment (jjaassoonn) [reply Mathlib/RingTheory/TwoSidedIdeal/Basic.lean]: removed

File: Mathlib/RingTheory/TwoSidedIdeal/Basic.lean (hunk 2 above)
Comment (jjaassoonn) [reply Mathlib/RingTheory/TwoSidedIdeal/Basic.lean]: changed `=>` to `\mapsto` throughout

**Comment by kbuzzard** [comment]
Thanks!

maintainer merge

**Comment by mattrobball** [comment]
!bench

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**